api/users/serializers.py

```diff
@@ -195,6 +198,16 @@ class ReadEmailUserDetailSerializer(UserDetailSerializer):
     email = ser.CharField(source='username', read_only=True)
 
 
+class UserMeDetailSerializer(ReadEmailUserDetailSerializer):
+
+    can_view_reviews = ser.SerializerMethodField(help_text='Whether the current user has the `view_submissions` permission to ANY reviews provider.')
+
+    def get_can_view_reviews(self, obj):
+        group_qs = GroupObjectPermission.objects.filter(group__user=obj, permission__codename='view_submissions')
+        user_qs = UserObjectPermission.objects.filter(user=obj, permission__codename='view_submissions')
+        return group_qs.exists() or user_qs.exists()
```

Comment:
How about using `guardian.shortcuts.get_objects_for_user(obj, 'view_submissions')`?

Reply:
It would be [significantly slower](https://github.com/django-guardian/django-guardian/blob/devel/guardian/shortcuts.py#L350).

It makes more queries than required.
It loads all the objects, which we don't need to.
It checks both the user and group permissions every time, 9/10 we'll only have to check the  group permissions.

Reply:
Hmm, good points, maybe I'll make a util somewhere instead of `get_objects_for_user`. I assumed it just constructed and returned a simple queryset, since that seems relatively easy/sane to do